src/app/providers/authentication/__tests__/authentication.spec.ts

```diff
@@ -0,0 +1,163 @@
+import { TestBed, async } from '@angular/core/testing';
+import { configureTestSuite } from 'ng-bullet';
+
+import { AuthenticationProvider, Token } from '../authentication';
+import { AppConfigProvider } from '../../app-config/app-config';
+import { AppConfigProviderMock } from '../../app-config/__mocks__/app-config.mock';
+import { NetworkStateProvider, ConnectionStatus } from '../../network-state/network-state';
+import { NetworkStateProviderMock } from '../../network-state/__mocks__/network-state.mock';
+import { AppConfig } from '../../app-config/app-config.model';
+import { DataStoreProvider } from '../../data-store/data-store';
+import { DataStoreProviderMock } from '../../data-store/__mocks__/data-store.mock';
+
+describe('Authentication', () => {
+  let authenticationProvider: AuthenticationProvider;
+  let networkStateProvider: NetworkStateProvider;
+  let appConfigProvider: AppConfigProvider;
+  let dataStoreProvider: DataStoreProvider;
+
+  configureTestSuite(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        AuthenticationProvider,
+        { provide: AppConfigProvider, useClass: AppConfigProviderMock },
+        { provide: NetworkStateProvider, useClass: NetworkStateProviderMock },
+        { provide: DataStoreProvider, useClass: DataStoreProviderMock },
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+    networkStateProvider = TestBed.inject(NetworkStateProvider);
+    authenticationProvider = TestBed.inject(AuthenticationProvider);
+    appConfigProvider = TestBed.inject(AppConfigProvider);
+    dataStoreProvider = TestBed.inject(DataStoreProvider);
+    authenticationProvider.initialiseAuthentication();
+  }));
+
+  describe('Provider', () => {
+
+    it('should compile', () => {
+      expect(authenticationProvider).toBeDefined();
+    });
+
+    it('determineAuthenticationMode() should set unauthenticated mode to true if offline', async () => {
+      spyOn(networkStateProvider, 'getNetworkState').and.returnValue(ConnectionStatus.OFFLINE);
+      authenticationProvider.determineAuthenticationMode();
+      expect(authenticationProvider.isInUnAuthenticatedMode()).toEqual(true);
+    });
+
+    it('determineAuthenticationMode() should set unauthenticated mode to false if online', async () => {
+      spyOn(networkStateProvider, 'getNetworkState').and.returnValue(ConnectionStatus.ONLINE);
+      authenticationProvider.determineAuthenticationMode();
+      expect(authenticationProvider.isInUnAuthenticatedMode()).toEqual(false);
+    });
```

Comment:
Do these two need to be `async` functions?

Reply:
removed the `async` from these tests